nanopc-t6.mk

```diff
@@ -0,0 +1,261 @@
+TPL_BIN			?= $(BINARIES_PATH)/rk3588_ddr_lp4_2112MHz_lp5_2400MHz_v1.18.bin
+BOOT_CMD_PATCH		?= $(ROOT)/build/nanopc-t6/nanopi6.h.patch
+BOARD_DTSI_PATCH	?= $(ROOT)/build/nanopc-t6/rk3588-nanopi6-common.dtsi.patch
+LINUX_DTSI		?= $(LINUX_PATH)/arch/arm64/boot/dts/rockchip/rk3588-nanopi6-common.dtsi
+UBOOT_HEADER		?= $(UBOOT_PATH)/include/configs/nanopi6.h
+
+LINUX_MODULES ?= n
+
+BR2_TARGET_ROOTFS_CPIO = n
+BR2_TARGET_ROOTFS_CPIO_GZIP = n
+BR2_TARGET_ROOTFS_EXT2 = y
+# Use Debug UART (tty1) for system console
+# Reference: https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T6#NanoPC-T6
+BR2_TARGET_GENERIC_GETTY_PORT = tty1
+ifeq ($(LINUX_MODULES),y)
+# If modules are installed...
+# ...enable automatic device detection and driver loading
+BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_EUDEV = y
+# ...and configure eth0 automatically based on ifup helpers
+BR2_PACKAGE_IFUPDOWN_SCRIPTS = y
+#BR2_SYSTEM_DHCP = eth0
+# An image with module takes more space
+BR2_TARGET_ROOTFS_EXT2_SIZE = 1536M
+# Enable SSH daemon for remote login
+BR2_PACKAGE_OPENSSH = y
+BR2_PACKAGE_OPENSSH_SERVER = y
+BR2_ROOTFS_POST_BUILD_SCRIPT = $(ROOT)/build/br-ext/board/nanopc-t6/post-build.sh
+# Enable NTP for current time
+BR2_PACKAGE_NTP = y
+BR2_PACKAGE_NTP_NTPD = y
+else
+BR2_TARGET_ROOTFS_EXT2_SIZE = 112M
+endif
+
+################################################################################
+# Targets
+################################################################################
+
+all: boot-img
+
+clean: buildroot-clean
+
+include toolchain.mk
+
+################################################################################
+# Arm Trusted Firmware-A
+################################################################################
+TF_A_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
+		M0_CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)"
+
+TF_A_DEBUG ?= $(DEBUG)
+ifeq ($(TF_A_DEBUG),0)
+TF_A_LOGLVL ?= 30
+TF_A_OUT = $(TF_A_PATH)/build/rk3588/release
+else
+TF_A_LOGLVL ?= 40
+TF_A_OUT = $(TF_A_PATH)/build/rk3588/debug
+endif
+
+TF_A_FLAGS ?= ARCH=aarch64 PLAT=rk3588 SPD=opteed DEBUG=$(TF_A_DEBUG) \
+	      LOG_LEVEL=$(TF_A_LOGLVL)
+
+.PHONY: tfa
+tfa:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) bl31
+
+.PHONY: tfa-clean
+tfa-clean:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) clean
+
+clean: tfa-clean
+
+################################################################################
+# U-Boot
+################################################################################
+UBOOT_DEFCONFIG_FILES := $(UBOOT_PATH)/configs/nanopi6_defconfig \
+			 $(ROOT)/build/kconfigs/u-boot_nanopc-t6.conf
+
+UBOOT_FLAGS ?= CROSS_COMPILE=$(CROSS_COMPILE_NS_KERNEL) \
+	       CC=$(CROSS_COMPILE_NS_KERNEL)gcc \
+	       HOSTCC="$(CCACHE) gcc"
+
+$(TPL_BIN):
+	mkdir -p $(BINARIES_PATH)
+	cd $(BINARIES_PATH) && \
+		wget -O $(notdir $(TPL_BIN)) https://github.com/rockchip-linux/rkbin/raw/master/bin/rk35/$(notdir $(TPL_BIN))
```

Comment:
Doesn't it work with `wget -O $(TPM_BIN) ...` without the `cd $(BINARIES_PATH)`?